File: leetcode/LeetCode_448_On_space.py
class Solution:
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        for i,x in enumerate(nums):
            index = abs(nums[i]) - 1
            nums[index] = -nums[index] if nums[index] > 0 else nums[index]

        ans = []
        for i,x in enumerate(nums):
            if nums[i] > 0:
                ans.append(i+1)
        return ans

File: leetcode_archived_cpp/LeetCode_448_On_space.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int len = nums.size();
        for(int i = 0; i < len;i++)
        {
            int index = abs(nums[i])-1;
            nums[index] = nums[index] > 0 ? -nums[index] : nums[index];
        }

        vector<int> ans;
        for(int i = 0;i < len; i++)
        {
            if (nums[i] > 0)
            {
                ans.push_back(i+1);
            }
        }
        return ans;
    }
};
